### ISSUE TITLE
Split op case not handled. Input shape = [1, 2600], output shape = [1, 650]

### ISSUE THREAD

**Issue by lngvietthang** [tf-coreml/tf-coreml#198]
This error happen when I try to convert my freeze graph to coreml model.
Split op is one ops of LSTMCell class.
I tested split ops in tfcoreml:
`pytest tests/test_tf_converter.py::TFSingleLayersTest::test_split`
result is success.
Anyone can give me some suggestion to fix this.
More details:
Output size of LSTM is 650 dim

Thanks./

**Comment by aseemw** [comment]
Actually there are two issues here:
- split op in CoreML only supports equal splitting. So this op is unsupported (an alternative is to use a custom layer). 

- tfcoreml converter does not support LSTMs. Actually I am surprised that you do not get an error saying that the graph has cycles.

**Comment by lngvietthang** [comment]
> tfcoreml converter does not support LSTMs. Actually I am surprised that you do not get an error saying that the graph has cycles.
I used static_rnn in my model so it generated unrolled graph for lstm so the graph did not contain any cycles inside

**Comment by aseemw** [comment]
I see. 
Then the issue is the first one I stated above.

**Comment by madnavs** [comment]
@lngvietthang I'm using static_rnn too and got the same error message. Did you write a custom layer for split_op? or any other trick?

**Comment by lngvietthang** [comment]
@madnavs No, I use BasicRNNCell instead of LSTMBlockCell, which does not have split_op :D

**Comment by madnavs** [comment]
@lngvietthang  Thanks that worked! 
But I got this 
![unnamed](https://user-images.githubusercontent.com/16426068/51003036-147ab680-14ea-11e9-83f8-e1b55b469909.png)

**Comment by Mahanotrahul** [comment]
Hello, I am also facing a similar problem while converting the model to coreml- 
`ValueError: Split op case not handled. Input shape = [1, 512], output shape = [1, 128]`

Is there anyway I can use tensorflow LSTMCell or an LSTM custom layer instead of BasicRNNCell?

**Comment by gordonrust** [comment]
@aseemw       i see the following condition checks in _shape_sensitive_layers.py in _add_split at lines 214-216.
   if not ((len(input_shape) == 4 and len(common_out_shape) == 4 and common_out_shape[:3] == input_shape[:3]) \
          or (len(input_shape) == 3 and len(common_out_shape) == 3 and common_out_shape[:2] == input_shape[:2])):
    raise ValueError('Split op case not handled. Input shape = {}, output shape = {}'.format(str(input_shape), str(common_out_shape)))

As far as I can tell,  u are simply ensuring that the dimesion u are splitting is the last dim. Why this constrain?
